### ISSUE TITLE
Clearing Number Range Filter

### ISSUE THREAD

**Issue by JohnHanks** [vedmack/yadcf#276]
Uncaught TypeError: Cannot read property 'replace' of undefined
    rangeClear	@	jquery.dataTables.yadcf.js:2716
    onclick	@	Devices.cfml:1

sometimes I get this error when I try to clear the number range filter.
the onclick is me clicking the number range filter reset x button.

the filter doesn't clear usually unless I click another text input before clicking the x button.

and what I just wrote, doesn't always work the first time, or the second time...

in short, the behavior is very sporadic, i can't quite quantify what is happening.

i will attempt to fiddle this.  if i cannot, i may need to give you login access to our app, if you are not opposed :smile:

btw using latest beta35 ver.

**Comment by vedmack** [comment]
Are you sure you are using the [0.8.9.beta.35](https://github.com/vedmack/yadcf/blob/1e23a16abf9f0419080e57905045a047e7eb43dc/jquery.dataTables.yadcf.js) , because there is no `repalce` usage at the mentioned lien of code, [see here line 2716](https://github.com/vedmack/yadcf/blob/1e23a16abf9f0419080e57905045a047e7eb43dc/jquery.dataTables.yadcf.js#L2716), if you are sure about the line of code/yadcf version please provide the link to your sire, otherwise let me know about the right version/line of code

**Comment by JohnHanks** [comment]
oops!

line 2706 is the actual line

**Comment by JohnHanks** [comment]
On further investigation, I think I have found at least one culprit.  I have a span inside of my button (so that I can use the glyphicon-remove icon instead of an ASCII "X")  and if I click on the span (the X) I get the console error logged.  If I click on the button space that does not include the span (like close to the edge/corner) then the filter clears as it should.  Maybe this has to do with the event.target portion

I noticed separately also that a bunch of divs are getting created with id="yadcf-filter-wrapper--filter_div_#"  where # can be all of my columns (lets say 7,8,9) and appended after the first div with id="yadcf-filter-wrapper--filter_div_#" that has the inputs inside of it being filtered.  These divs are being created when I type in any of my range number filter boxes and also when I clear my filter successfully (clicking the button, but not the span in the button).

**Comment by vedmack** [comment]
I changed the way I'm getting the column number in that function, and that should work for all types of reset buttons of a range filter, grab the lateset version [0.8.9.beta.37](https://github.com/vedmack/yadcf/blob/999fc40ad1c24ed2b1349decc0edbef9d11b56a6/jquery.dataTables.yadcf.js) , if you will bump into similar cases let me know in this issue.

**Comment by JohnHanks** [comment]
I see that the error went away in the console, but the same applies.  No ability to trigger my button, and the extra divs are there.
